src/arrayMethodSort.js

```diff
@@ -4,8 +4,38 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compare =
+      typeof compareFunction === 'function'
+        ? compareFunction
+        : (a, b) => {
+          const strA = String(a);
+          const strB = String(b);
+
+          if (strA < strB) {
+            return -1;
+          }
+
+          if (strA > strB) {
+            return 1;
+          }
+
+          return 0;
+        };
+
+    const arr = this;
+    const len = arr.length;
+
+    for (let i = 0; i < len - 1; i++) {
+      for (let j = 0; j < len - i - 1; j++) {
+        if (compare(arr[j], arr[j + 1]) > 0) {
+          // Swap elements if they are in the wrong order
+          [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
+        }
+      }
```

Comment:
The nested for-loops correctly implement the bubble sort algorithm. Just be aware that this is a naive implementation and there are optimizations possible, such as stopping the algorithm if no elements were swapped during a pass, which indicates that the array is already sorted.